### ISSUE TITLE
WEP cracking not triggered beyond 10000 IVs

### ISSUE THREAD

**Issue by OscarAkaElvis** [aircrack-ng/aircrack-ng#2018]
# Issue type

- Defect - Unexpected behavior (obvious or confirmed in the forum)

# System information

- OS: Parrot or Kali, not relevant
- CPU: not relevant
- Wireless card and chipset (if relevant): tested using ralink, atheros, not relevant.

# Aircrack-ng version

- Version: Error reproduced using 1.5.2-3 . Working flawlessly on 1.2-rc4
The error could be on any revision between them.

# Defect
## How to reproduce the issue

Just launch an airodump process to capture IVs for WEP network while in other trying to crack it. The expected behavior is aircrack starting to crack when 5000 IVs are reach.  It starts at 5000. obviously, it can't crack it using only 5000... then it is suppossed that it will try again on 10000 IVs, but on new 1.5.2 this never happens. The cracking proccess stuck waiting for 10000 ivs ... even if you capture 50000 IVs. It works on 1.2 rc4 flawlessly.

I'll put a couple of screenshots from the [airgeddon](https://github.com/v1s1t0r1sh3r3/airgeddon) script. The use of airgeddon here is not important, this can be done manually, I just detected the bug using it.

![aircrack-1 5 2-3_wep-not-working](https://user-images.githubusercontent.com/5803348/52155840-21bc3880-2685-11e9-859f-cad95ea5fa56.png)

![aircrack-1 2-rc4_wep-working](https://user-images.githubusercontent.com/5803348/52155850-2da7fa80-2685-11e9-9cfd-e783ccc7f6ff.png)

**Comment by OscarAkaElvis** [comment]
From which aircrack version is this fixed? Thanks

**Comment by OscarAkaElvis** [comment]
More than a year and this is still not fixed. You closed the issue... but it is not working yet. I checked it again on aircrack 1.6

![image](https://user-images.githubusercontent.com/5803348/79658427-b7bd7f80-81b3-11ea-8197-92a82fe0818e.png)

It's a pity to lost old WEP features. I know WEP is very old, but sometimes we can see a network of this type, at least in my country. It is rare... but sometimes happen. Please, could anybody take charge of this? It could be awesome. Thanks so much.

**Comment by aircrack-ng** [comment]
More than 10K data doesn't always mean more than 10K unique IV. It likely isn't  the case. If you believe it is wrong, open a new bug with the PCAP attached.

**Comment by OscarAkaElvis** [comment]
Hi, no need of a new issue I think, we can do it here. If you prefer I can open a new issue but if is the same issue we can discuss here about it. Don't worry, I'll put the links to the files.

I tested it further doing some tests... and almost all the times stuck at 10000 IVs as always... but... once, i passed from 10000 to 15000 and then it stucked there. ¿? I think aircrack is doing something bad counting IVs....

![15000ivs](https://user-images.githubusercontent.com/5803348/79797983-892cd980-8358-11ea-91d1-ac69ce928d9c.png)

Then, during this same attack I decided to save the cap at some point (near 30000 IVs) doing a copy of the file to another location while the attack is still running, and i have a resulting file containing about 30000 IVs. This is the link to download the file (don't care too much about personal data or if you see the password, is a testing AP): https://mega.nz/file/8NIlESZL#VAPChhqio3_73lkPnXHQK_UQpYd7Cu7SdoHeVHUDgbk

Launching the decrypt attack command over it, `aircrack-ng wepdata-01.cap` it decrypts the password pretty quickly as expected.

![cracked](https://user-images.githubusercontent.com/5803348/79798380-330c6600-8359-11ea-978b-a7dc7998471c.png)

Then, during the same attack... much more IVs were captured... as you can see here and still stuck on 15000:

![stuck](https://user-images.githubusercontent.com/5803348/79798990-13297200-835a-11ea-8437-260c766a17e3.png)

I saved again the file a little bit after the screenshot... so in screenshot it shows about 57000 IVs but this second file contains about 67000 IVs: https://mega.nz/file/oNo1GRIL#1fgWcJOVEVojDahupNMAkf2NRXLR08GFIUcgxwvR0NQ

The curious thing is that using this file which contains much more IVs, aircrack is not able to crack it.... ¿¿??? look:

![notcracked](https://user-images.githubusercontent.com/5803348/79799496-d14cfb80-835a-11ea-9ed6-5b630f3211d2.png)

What the hell is going on here? why is able to crack it containing about 30000 IVs and is not able to do it having near 70000 IVs? anyway, this is just curiosity. The bug is that aircrack is not counting well received IVs to relaunch decrypt over the file.

All tests were done using same aircrack-ng version. 1.6 which is the latest right now at Parrot repositories. Remember that on aircrack-ng 1.2 this is working like a charm as I reported on first post of this thread (maybe it could be useful to you to search through commits if needed to check changes done).

Thanks so much.

**Comment by xpz3** [comment]
@jbenden @Mister-X- @aircrack-ng 
[Fix 1 - ignore read error on packet reader thread and retry](https://github.com/xpz3/aircrack-ng/commit/ba3c58704e3e54a5c0ff064877d80e0d2562613b)

The above commit fixes the issue in this thread. I'm no crypto expert and so the commits above are purely based on code flow analysis. There may be other methods to achieve the same behaviour. I think i know one more method but to implement it, i need to study some more on a function in aircrack. I've already spent much time on this even though it's pretty old. But i did this so that you guys can pin point the exact issue and we @ airgeddon can have benefit too if you decide to fix this at last!!! Thanks
Edit: Should i post this as a new issue? I really don't know...
Edit 2: I removed the second commit to create a pull request
